math.Pow returns strange (and very wrong) result for 0.5^1

Test case:

```
func TestFirstPowers(t *testing.T) {
	one := decimal.New(1, 0)
	for i := int64(0); i < 10; i++ {
		n := decimal.New(i, 1)
		firstPow := Pow(new(decimal.Big), n, one)
		if n.Cmp(firstPow) != 0 {
			t.Errorf("%v^1 != %v", n, firstPow)
		} else {
			t.Logf("%v^1 == %v", n, firstPow)
		}
	}
}
```

Outputs:

```
=== RUN   TestFirstPowers
--- FAIL: TestFirstPowers (0.00s)
    pow_test.go:17: 0.0^1 == 0
    pow_test.go:17: 0.1^1 == 0.1
    pow_test.go:17: 0.2^1 == 0.2
    pow_test.go:17: 0.3^1 == 0.3
    pow_test.go:17: 0.4^1 == 0.4
    pow_test.go:15: 0.5^1 != 0.7071067811865475
    pow_test.go:17: 0.6^1 == 0.6
    pow_test.go:17: 0.7^1 == 0.7
    pow_test.go:17: 0.8^1 == 0.8
    pow_test.go:17: 0.9^1 == 0.9
```

Looks like 0.5^1 gives sqrt(2)/2, which doesn't make a hell of a lot of sense but here we are.

--- a/math/math_test.go
+++ b/math/math_test.go
@@ -3,7 +3,9 @@ package math_test
 import (
 	"testing"
 
+	"github.com/ericlagergren/decimal"
 	"github.com/ericlagergren/decimal/internal/test"
+	"github.com/ericlagergren/decimal/math"
 )
 
 func TestExp(t *testing.T)   { test.Exp.Test(t) }
@@ -11,3 +13,16 @@ func TestLog(t *testing.T)   { test.Log.Test(t) }
 func TestLog10(t *testing.T) { test.Log10.Test(t) }
 func TestPow(t *testing.T)   { test.Pow.Test(t) }
 func TestSqrt(t *testing.T)  { test.Sqrt.Test(t) }
+
+func TestIssue146(t *testing.T) {
+	one := decimal.New(1, 0)
+	for i := int64(0); i < 10; i++ {
+		n := decimal.New(i, 1)
+		firstPow := math.Pow(new(decimal.Big), n, one)
+		if n.Cmp(firstPow) != 0 {
+			t.Errorf("%v^1 != %v", n, firstPow)
+		} else {
+			t.Logf("%v^1 == %v", n, firstPow)
+		}
+	}
+}

--- a/math/pow.go
+++ b/math/pow.go
@@ -54,7 +54,7 @@ func Pow(z, x, y *decimal.Big) *decimal.Big {
 		return z.SetUint64(1)
 	}
 
-	if x.Cmp(ptFive) == 0 {
+	if y.Cmp(ptFive) == 0 {
 		// x ** 0.5 = sqrt(x)
 		return Sqrt(z, x)
 	}

